src/main/java/com/aws/greengrass/util/ProxyUtils.java

```diff
@@ -208,20 +238,62 @@ public static SdkHttpClient getSdkHttpClient() {
     /**
      * <p>Boilerplate for providing a proxy configured ApacheHttpClient builder to AWS SDK v2 client builders.</p>
      *
+     * <p>To support HTTPS proxies and other scenarios, the HttpClient is configured with a trust manager
+     * containing the root CAs from the nucleus configuration and the JVM's default root CAs.</p>
+     *
      * <p>If you need to customize the HttpClient, but still need proxy support, use <code>ProxyUtils
      * .getProxyConfiguration()</code></p>
      *
-     * @return httpClient built with a ProxyConfiguration or null if no proxy is configured (null is ignored in AWS
-     *     SDK clients)
+     * @return httpClient builder with a ProxyConfiguration, if a proxy is configured, otherwise
+     *         a default httpClient builder
      */
     public static ApacheHttpClient.Builder getSdkHttpClientBuilder() {
         ProxyConfiguration proxyConfiguration = getProxyConfiguration();
+        TrustManager[] trustManagers = createTrustManagers();
 
         if (proxyConfiguration != null) {
-            return ApacheHttpClient.builder().proxyConfiguration(proxyConfiguration);
+            return ApacheHttpClient.builder()
+                    .tlsTrustManagersProvider(() -> trustManagers)
```

Comment:
just pass ProxyUtils::createTrustManager instead of a lambda returning the local thing

Reply:
Done